### PR TITLE
Diagnostics for `QuietDownCommandTest.quietDownShouldSuccessWithBlockPlusNonExpiredTimeoutAndRunningExecutor`

### PR DESCRIPTION
**Pull request by jglick**
Alternative diagnostics building on #8513; see https://github.com/jenkinsci/jenkins-test-harness/pull/657#issuecomment-1739028388 with @Vlatombe & @basil

### Testing done

Test passes locally and shows

```
…
   9.481 [aProject #1] Finished: SUCCESS
   9.772 [id=20]	INFO	hudson.lifecycle.Lifecycle#onStatusUpdate: Stopping Jenkins
   9.828 [id=20]	FINEST	hudson.XmlFile#write: Writing …/queue.xml
java.lang.Throwable
	at hudson.XmlFile.write(XmlFile.java:205)
	at hudson.model.Queue.save(Queue.java:472)
	at jenkins.model.Jenkins._cleanUpPersistQueue(Jenkins.java:3921)
	at jenkins.model.Jenkins.cleanUp(Jenkins.java:3676)
	at org.jvnet.hudson.test.JenkinsRule._stopJenkins(JenkinsRule.java:566)
	at org.jvnet.hudson.test.JenkinsRule.after(JenkinsRule.java:514)
	at org.jvnet.hudson.test.JenkinsRule$1.evaluate(JenkinsRule.java:625)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:829)
WARN: The method class org.apache.commons.logging.impl.SLF4JLogFactory#release() was invoked.
WARN: Please see http://www.slf4j.org/codes.html#release for an explanation.
   9.842 [id=20]	INFO	hudson.lifecycle.Lifecycle#onStatusUpdate: Jenkins stopped
   9.889 [id=20]	INFO	o.j.h.t.TemporaryDirectoryAllocator#dispose: deleting …
```

as expected. Unknown if the culprit would in fact appear during a flake.

### Proposed changelog entries

- N/A

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

File: test/src/test/java/hudson/cli/QuietDownCommandTest.java
Comment (jglick) [inline test/src/test/java/hudson/cli/QuietDownCommandTest.java:359]: Among other :raised_eyebrow: in this test, this variable was unused.

**Comment by Vlatombe** [comment]
Thought of this though I wanted to have a generic option because this is typically the kind of failure appearing on Windows for reasons that can be unrelated to the test itself.

**Comment by jglick** [comment]
https://github.com/jenkinsci/jenkins/pull/8511/checks?check_run_id=17000216956 happened to be on Windows, perhaps just because Windows CI tends to be slower and more likely to expose race conditions, but I strongly suspect the flake in this case is the fault of the test.

**Review by timja — APPROVED** [review]

**Review by Vlatombe — APPROVED** [review]

**Review by NotMyFault — APPROVED** [review]
/label ready-for-merge

---

This PR is now ready for merge. We will merge it after ~24 hours if there is no negative feedback.
Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process.
Thanks!

**Comment by jglick** [comment]
https://github.com/jenkinsci/jenkins/pull/8640/checks?check_run_id=18102119192 :eyes:

**Pull request merged**
